### ISSUE TITLE
On failure, x:expect with context reports the entire result, not the specified context

### ISSUE THREAD

**Issue by AirQuick** [xspec/xspec#33]
**test.xsl** (just returns the input node intact)
```xml
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet exclude-result-prefixes="#all" version="2.0"
	xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
	<xsl:template as="node()" match="node()">
		<xsl:sequence select="." />
	</xsl:template>
</xsl:stylesheet>
```

**test_old.xspec** (tests **test.xsl** by looking for `descendant::baz`)
```xml
<?xml version="1.0" encoding="UTF-8"?>
<x:description stylesheet="test.xsl" xmlns:x="http://www.jenitennison.com/xslt/xspec">
	<x:scenario label="Scenario-old">
		<x:context><foo><bar><baz>original-text</baz></bar></foo></x:context>
		<x:expect label="Expect success" test="descendant::baz"><baz>original-text</baz></x:expect>
		<x:expect label="Expect failure" test="descendant::baz"><baz>broken-text</baz></x:expect>
	</x:scenario>
</x:description>
```

`x:expect label="Expect failure"` is expected to fail, because of the text node difference.

With XSpec 0.3.0, the failure report HTML is generated as follows. (Indent and xmlns omitted for brevity)

Result | Expecting
------ | ---------
`<baz>original-text</baz>` | `<baz>broken-text</baz>`

So far, so good.

XSpec 0.4.0-RC changed the recommended way to describe the same `x:expect`. (Just my guess... Any clarification or documentation welcomed.)

**test_new.xspec** (`test` attribute in **test_old.xspec** is replaced with `context` attribute)
```xml
<?xml version="1.0" encoding="UTF-8"?>
<x:description stylesheet="test.xsl" xmlns:x="http://www.jenitennison.com/xslt/xspec">
	<x:scenario label="Scenario-new">
		<x:context><foo><bar><baz>original-text</baz></bar></foo></x:context>
		<x:expect label="Expect success" context="descendant::baz"><baz>original-text</baz></x:expect>
		<x:expect label="Expect failure" context="descendant::baz"><baz>broken-text</baz></x:expect>
	</x:scenario>
</x:description>
```

Running this scenario with the latest master, the failure report HTML is generated as follows. (Indent omitted for brevity)

Result | Expecting
------ | ---------
`<foo><bar><baz>original-text</baz></bar></foo>` | `<baz>broken-text</baz>`

Notice the left-hand **Result** is the entire result, not `<baz>original-text</baz>` which 0.3.0 used to report.

**Comment by cirulls** [comment]
Thanks for reporting this issue, I'll have a look and keep you posted. Please allow me 7-10 days to get back to you as it is Christmas time.

**Comment by cirulls** [comment]
I merged #38 and close this issue.
